test/gateway/schema.js

```diff
@@ -0,0 +1,265 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const GQL = require('../..')
+
+async function createService (t, port, schema, resolvers = {}) {
+  const service = Fastify()
+  t.tearDown(() => service.close())
+  service.register(GQL, {
+    schema,
+    resolvers,
+    federationMetadata: true
+  })
+  await service.listen(port)
+}
+
+test('It builds the gateway schema correctly', async (t) => {
+  const users = {
+    u1: {
+      id: 'u1',
+      name: 'John'
+    },
+    u2: {
+      id: 'u2',
+      name: 'Jane'
+    },
+    u3: {
+      id: 'u3',
+      name: 'Jack'
+    }
+  }
+
+  const posts = {
+    p1: {
+      pid: 'p1',
+      title: 'Post 1',
+      content: 'Content 1',
+      authorId: 'u1'
+    },
+    p2: {
+      pid: 'p2',
+      title: 'Post 2',
+      content: 'Content 2',
+      authorId: 'u2'
+    },
+    p3: {
+      pid: 'p3',
+      title: 'Post 3',
+      content: 'Content 3',
+      authorId: 'u1'
+    },
+    p4: {
+      pid: 'p4',
+      title: 'Post 4',
+      content: 'Content 4',
+      authorId: 'u2'
+    }
+  }
+
+  await createService(t, 4001, `
```

Comment:
Please avoid using defined ports in the tests. Those should not be set (or set to `0`), and you should get the port or the URL as the result of `createService`.